pkg/event/consumer.go

```diff
@@ -0,0 +1,216 @@
+		sm    schema.SchemaMap
+		s     *schema.Schema
+	)
+	if msg == nil {
+		return fmt.Errorf("msg cannot be nil")
+	}
+	if event, err = getHeader(msg, string(message.HdrType)); err != nil {
+		return fmt.Errorf("header '%s' not found: %s", string(message.HdrType), err.Error())
+	}
+	if !isValidEvent(string(event.Value)) {
+		return fmt.Errorf("event not valid: %v", event)
+	}
+	if msg.TopicPartition.Topic == nil {
+		return fmt.Errorf("topic cannot be nil")
+	}
+	topic := *msg.TopicPartition.Topic
+	if sm = getSchemaMap(schemas, topic); sm == nil {
+		return fmt.Errorf("topic '%s' not found in schema mapping", event.String())
+	}
+	if s = getSchema(sm, string(event.Value)); s == nil {
+		return fmt.Errorf("schema '%s'  not found in schema mapping", event.String())
+	}
+
+	return s.ValidateBytes(msg.Value)
+}
+
+func logEventMessageError(msg *kafka.Message, err error) {
+	if msg == nil {
+		log.Logger.Error().Msgf("msg is nil")
+		return
+	}
+	if err == nil {
+		log.Logger.Error().Msgf("err is nil")
+		return
+	}
+	log.Logger.Error().
+		Msgf("error processing event message: headers=%v; payload=%v: %s", msg.Headers, string(msg.Value), err.Error())
+}
+
+func NewConsumerEventLoop(consumer *kafka.Consumer, handler Eventable) func() {
+	var (
+		err     error
+		msg     *kafka.Message
+		schemas schema.TopicSchemas
+	)
+	sigchan := make(chan os.Signal, 1)
+	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)
+	if schemas, err = schema.LoadSchemas(); err != nil {
+		panic(err)
+	}
+
+	return func() {
+		log.Logger.Info().Msg("Consumer loop awaiting to consume messages")
+		for {
+			// Message wait loop
+			for {
+				msg, err = consumer.ReadMessage(1 * time.Second)
+
+				if err != nil {
+					val, ok := err.(kafka.Error)
+					if !ok || val.Code() != kafka.ErrTimedOut {
+						log.Logger.Error().Msgf("error awaiting to read a message: %s", err.Error())
+					}
+
+					// TODO If moving into a multi-service design
+					//      this signal handler can not be here
+					//      and potentially the graceful termination
+					//      would need some communication with the
+					//      context (<-context.Done())
+					//
+					// select {
+					// case <-ctx.Done():
+					// 	 log.Logger.Info().Msgf("Stopping consumer event loop")
+					// 	 return
+					// default:
+					// }
+					//
+					// Keep in mind that VSCode go debugger send a SIGKILL
+					// signal to the debugged process, which cannot be captured
+					// and no graceful termination could happen.
+					// https://github.com/golang/vscode-go/issues/120#issuecomment-1092887526
+					//
+					// The multi-service design could comes when adding
+					// instrumentation which will public endpoints for
+					// collecting prometheus metrics.
+					select {
+					case sig := <-sigchan:
+						log.Logger.Info().Msgf("Caught signal %v: terminating\n", sig)
+						return
+					default:
+					}
+
+					continue
+				}
+				break
+			}
+
+			if err = validateMessage(schemas, msg); err != nil {
```

Comment:
TODO Add informative log message that display: the kafka topic (source of the message), consumer group-id, key and X-Rh-Insights-Request-Id head (the last one to track along the system); that would be useful for troubleshooting.